server/app/controllers/obs.py

```diff
@@ -15,13 +18,81 @@ def obs(obs_id: ObservationId = None, limit_and_offset = None):
     with repository.transaction():
         observation = repository.read_observation(obs_id)
 
+        orbit = None
+
         if observation is None:
             abort(404, "Observation not found")
-    
+
+        orbit = observation
+        if observation['tle'] is not None:
+            orbit = parse_tle(observation['tle'][0], observation['tle'][1])
+
         files = repository.read_observation_files(observation["obs_id"],
             **limit_and_offset)
         files_count = repository.count_observation_files(obs_id)
         satellite = repository.read_satellite(observation["sat_id"])
 
+    # Now tweak some observation parameters to make them more human readable
+    observation = human_readable_obs(observation)
+
     return 'obs.html', dict(obs = observation, files=files,
-        sat_name=satellite["sat_name"], item_count=files_count)
+        sat_name=satellite["sat_name"], item_count=files_count, orbit=orbit)
+
+def parse_tle(tle1: str, tle2: str) -> dict:
+    """ Parses orbital data in TLE format and returns a dictionary with printable orbital elements
+        and other parameters."""
+
+    # First, parse the TLE lines. We don't care about the name.
+    t = TLE.from_lines(line1=tle1, line2=tle2, name="whatevah")
+
+    # Now convert it to poliastro orbit. All the data is there, but we want to
+    # make it easier to read, format it nicely and do some basic calculations.
+    # Hence the orb dictionary.
+    o = t.to_orbit()
+
+    RE = o.attractor.R # Earth radius
+    r_a = o.r_a - RE # Calculate apogee and perigee as altitude above Earth surface,
+    r_p = o.r_p - RE # rather than as distance from barycenter.
+
+    # Now make the parameters easier to read (cut unnecessary digits after comma, show altitude, etc)
+    orb = {}
+    orb["overview"] = repr(o)
+    orb["inc"] = "%4.1f %s" % (o.inc.value, o.inc.unit)
+    orb["a"] = o.a
+    orb["ecc"] = o.ecc
+    orb["r_a"] = "%4.1f %s (%4.1f %s above surface)" % (o.r_a.value, o.r_a.unit, r_a.value, r_a.unit)
+    orb["r_p"] = "%4.1f %s (%4.1f %s above surface)" % (o.r_p.value, o.r_p.unit, r_p.value, r_p.unit)
+    orb["raan"] = "%4.1f %s" % (o.raan.value, o.raan.unit)
+    orb["period"] = "%4.0f %s" % (o.period.value, o.period.unit)
+
+    # This is ugly hack. But it converts the ISO 8601 notation to a more human readable form.
+    tmp = str(o.epoch)[:19]
```

Comment:
If it is ugly for you then you can try
```o.epoch.to_datetime().isoformat()```.
Poliastro has own time structure. I don't have any experience with it then it may be possible do it a shorter way.

You can also use ```o.epoch.strftime``` and provide your own format.

Reply:
Used `o.epoch.strftime` as suggested.